src/main/java/com/ashkan/moviesapi/services/implementations/MovieRentalServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+import com.ashkan.moviesapi.exceptions.NotFound.MovieRentalNotFoundException;
+import com.ashkan.moviesapi.services.interfaces.MovieRentalService;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.util.List;
+
+@Service
+public class MovieRentalServiceImpl implements MovieRentalService {
+    private MovieRentalRepository movieRentalRepository;
+    private MovieRepository movieRepository;
+    private MemberRepository memberRepository;
+    private MovieCatalogRepository movieCatalogRepository;
+
+    @Autowired
+    public MovieRentalServiceImpl(MovieRentalRepository movieRentalRepository,
+                                  MovieRepository movieRepository,
+                                  MemberRepository memberRepository,
+                                  MovieCatalogRepository movieCatalogRepository) {
+        this.movieRentalRepository = movieRentalRepository;
+        this.movieRepository = movieRepository;
+        this.memberRepository = memberRepository;
+        this.movieCatalogRepository = movieCatalogRepository;
+    }
+
+    @Override
+    public List<MovieRental> findAll() {
+        return movieRentalRepository.findAll();
+    }
+
+    @Override
+    public MovieRental findById(int id) {
+        return movieRentalRepository.findById(id)
+                .orElseThrow(() -> new MovieRentalNotFoundException(id));
+    }
+
+    @Override
+    public void save(MovieRental movieRental) {
+        Movie movie = movieRepository.findById(movieRental.getMovieId())
+                .orElseThrow(() -> new MovieRentalNotFoundException(movieRental.getMovieId()));
+
+        Member member = memberRepository.findById(movieRental.getMemberId())
+                .orElseThrow(() -> new MovieRentalNotFoundException(movieRental.getMemberId()));
+
+        MovieCatalog movieCatalog = movieCatalogRepository.findById(movieRental.getMovieId())
+                .orElseThrow(() -> new RuntimeException("Movie catalog not found " + movieRental.getMovieId()));
+
+        Timestamp currentDate = new Timestamp(System.currentTimeMillis());
+        Timestamp returnDate = new Timestamp(System.currentTimeMillis() + constants.DAYS_TO_ADD);
+
+        movieRental.setMovieByMovieId(movie);
+        movieRental.setMemberByMemberId(member);
+        movieRental.setDate(currentDate);
+        movieRental.setToReturnDate(returnDate);
+        movieRental.setStatus("Rented");
+
+        if (movieCatalog.getCopies() > 0) {
+            movieCatalog.setCopies(movieCatalog.getCopies() - 1);
+        } else {
+            throw new MovieRentalBadRequestException_NoCopies();
+        }
+
+        movieRentalRepository.save(movieRental);
+    }
+
+    @Override
+    public void deleteById(int id) {
+        movieRentalRepository.deleteById(id);
+    }
+
+    @Override
+    public void patchStatus(MovieRental movieRental, int id) {
```

Comment:
Since the requirements only ask to have the status as part of the payload you can define a new object with that property directly instead of using the entire Entity because the client is able to send the other properties and the client may think his changes will be taked into account.
Just think about it please. :)

Reply:
I think this problem should be solved with the DTOs problem